src/v/kafka/server/protocol.cc

```diff
@@ -95,10 +96,35 @@ coordinator_ntp_mapper& protocol::coordinator_mapper() {
 }
 
 config::broker_authn_method get_authn_method(const net::connection& conn) {
+    // Global enable_sasl is deprecated, replaced by broker_authn_method == sasl
```

Comment:
I don't know if we can really say enable_sasl is deprecated: from the if/else below, it looks like the only ways to enable sasl are either:
- Use `enable_sasl`
- Modify node configuration.

If we're serious about deprecating enable_sasl (i.e. really removing it eventually), then that would mean the end state is that the only way to switch on SASL is to modify node configuration, which feels like a regression in UX for users who don't care about mTLS: they lose the single central switch for SASL.

I think we should either _not_ deprecate enable_sasl, or provide a way to centrally enable sasl via the enable_authorization setting: e.g. if they set enable_authorization but don't set mtls mapping rules, then sasl is switched on (for endpoints that do not have a per-endpoint setting in the node config)